Offline/OfflinePlayer/objc/OfflinePlayer/VideosViewController.m

```diff
@@ -0,0 +1,1370 @@
+                                                                 UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"Video Download Error"
+                                                                                                                                message:alertMessage
+                                                                                                                         preferredStyle:UIAlertControllerStyleAlert];
+                                                                 UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault
+                                                                                                                       handler:^(UIAlertAction * action) {}];
+                                                                 
+                                                                 [alert addAction:defaultAction];
+                                                                 [self presentViewController:alert animated:YES completion:nil];
+                                                             }
+                                                             
+                                                         });
+                                                         
+                                                     }];
+             }
+         });
+         
+     }];
+}
+
+#pragma mark - UITableView methods
+
+// Play the video in this row when selected
+- (IBAction)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(nonnull NSIndexPath *)indexPath
+{
+    NSDictionary *videoDictionary = self.videosTableViewData[ (int)indexPath.row ];
+    BCOVVideo *video = videoDictionary[@"video"];
+
+    if (video != nil)
+    {
+        [self.playbackController setVideos:@[ video ]];
+    }
+}
+
+- (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section
+{
+    return self.currentPlaylistTitle;
+}
+
+- (NSString *)tableView:(UITableView *)tableView titleForFooterInSection:(NSInteger)section
+{
+    return self.currentPlaylistDescription;
+}
+
+// Populate table with data from videosTableViewData
+- (UITableViewCell *)tableView:(UITableView *)tableView
+         cellForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    int row = (int)indexPath.row;
+    NSDictionary *videoDictionary = self.videosTableViewData[row];
+    BCOVVideo *video = videoDictionary[@"video"];
+    NSString *videoID = video.properties[@"id"];
+
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"video_cell"
+                                                            forIndexPath:indexPath];
+    cell.textLabel.text = video.properties[@"name"];
+    cell.textLabel.textColor = (video.usesFairPlay ? [UIColor colorWithRed:0.75 green:0.0 blue:0.0 alpha:1.0] : UIColor.blackColor);
```

Comment:
Add same comment as above about red for FairPlay.

Reply:
√